Source/Scene/ShadowMap.js

```diff
@@ -0,0 +1,397 @@
+                '    float shadow = texture2D(czm_shadowMapTexture, v_textureCoordinates).r; \n' +
+                '    gl_FragColor = vec4(shadow, shadow, shadow, 1.0); \n' +
+                '} \n';
+
+            // Set viewport now to avoid using a cached render state
+            var renderState = RenderState.fromCache({
+                viewport : new BoundingRectangle(x, y, width, height)
+            });
+
+            var drawCommand = frameState.context.createViewportQuadCommand(fs);
+            drawCommand.renderState = renderState;
+            drawCommand.pass = Pass.OVERLAY;
+            shadows._debugShadowViewCommand = drawCommand;
+        }
+
+        var viewport = shadows._debugShadowViewCommand.renderState.viewport;
+        viewport.x = x;
+        viewport.y = y;
+        viewport.width = width;
+        viewport.height = height;
+
+        frameState.commandList.push(shadows._debugShadowViewCommand);
+    }
+
+    function applyDebugSettings(shadowMap, frameState) {
+        if (!shadowMap.debugShow) {
+            return;
+        }
+
+        // TODO : will need to reposition if the light moves
+        if (!defined(shadowMap._debugLightFrustum)) {
+            var frustum = shadowMap._frustum;
+            var minimum = new Cartesian3(frustum.left, frustum.bottom, -frustum.near);
+            var maximum = new Cartesian3(frustum.right, frustum.top, -frustum.far);
+            var modelMatrix = shadowMap.camera.inverseViewMatrix;
+
+            // Create the light frustum
+            shadowMap._debugLightFrustum = new Primitive({
+                geometryInstances : new GeometryInstance({
+                    geometry : new BoxOutlineGeometry({
+                        minimum : minimum,
+                        maximum : maximum
+                    }),
+                    modelMatrix : modelMatrix,
+                    attributes : {
+                        color : ColorGeometryInstanceAttribute.fromColor(Color.BLUE)
+                    }
+                }),
+                appearance : new PerInstanceColorAppearance({
+                    translucent : false,
+                    flat : true
+                }),
+                asynchronous : false
+            });
+
+            // Create the light
+            shadowMap._debugLight = new Primitive({
+                geometryInstances : new GeometryInstance({
+                    geometry : new SphereOutlineGeometry({
+                        radius : 1.0
+                    }),
+                    modelMatrix : modelMatrix,
+                    attributes : {
+                        color : ColorGeometryInstanceAttribute.fromColor(Color.BLUE)
+                    }
+                }),
+                appearance : new PerInstanceColorAppearance({
+                    translucent : false,
+                    flat : true
+                }),
+                asynchronous : false
+            });
+
+            //// Create the camera frustum
```

Comment:
Do you still need this commented out code?  It is fine to leave it until this goes into master.